bytestring_splitter/__init__.py

```diff
@@ -44,18 +48,33 @@ def set_original_bytes_repr(self, bytes_representation):
         self._original_bytes = bytes_representation
 
     def finish(self):
-        finished_values = {}
+
         for message_name, (bytes_for_message, message_class, kwargs) in self.processed_objects.items():
-            finished_values[message_name] = produce_value(message_class,
+            self._finished_values[message_name] = produce_value(message_class,
                                                           message_name,
                                                           bytes_for_message,
                                                           kwargs)
-        return self._receiver(**finished_values, **self._additional_kwargs)
+        return self._receiver(**self._finished_values, **self._additional_kwargs)
+
+    def __getattr__(self, message_name):
+        # First we'll try to see if this message_name already has a finished value:
+        try:
+            return self._finished_values[message_name]
+        except KeyError:  # suppress might be good here, but it appears to have a performance penalty, and this is a performance-concerned function.
+            pass
+
+        try:
+            bytes_for_message, message_class, kwargs = self.processed_objects[message_name]
+            self._finished_values[message_name] = produce_value(message_class,
+                                                               message_name,
+                                                               bytes_for_message,
+                                                               kwargs)
+            produced_value = produce_value(message_class, message_name, bytes_for_message, kwargs)
+            del self.processed_objects[message_name]  # We don't do this as a pop() in case produce_value raises.
```

Comment:
I don't understand this comment, can you elaborate?

Reply:
It'd be marginally more optimal to do `self.processed_objects.pop(message_name)`, but in the (rare, but not unsupported) case that something changes in the receiver class such that `produce_value` raises at this time (but is caught and subsequently proceeds successfully), we'd already have erased it from `processed_objects`.